### DISCUSSION TITLE
InitiateSrpAuthRequest class not defined error

### DISCUSSION THREAD

**Discussion by Sourabh-ALTEN** [aws/aws-sdk-net-extensions-cognito#125] [Q&A] [answered]
Hello,

I am using this library to athenticate cognito user pool and have adpated it to my requirement. Below is the source code for the same

```
using Amazon;
using Amazon.Extensions.CognitoAuthentication;
using Amazon.CognitoIdentityProvider;
using Amazon.CognitoIdentityProvider.Model;

namespace CognitoJwtBearerTokenGenerator
{
    class Program
    {
        static async Task Main(string[] args)
        {
            if (args.Length < 5)
            {
                Console.WriteLine("Usage: dotnet run -- <username> <password> <clientId> <poolId> <region>");
                return;
            }

            var username = args[0];
            var password = args[1];
            .
            .
            
            var provider = new AmazonCognitoIdentityProviderClient(new Amazon.Runtime.AnonymousAWSCredentials(), Amazon.RegionEndpoint.GetBySystemName(region));
            var userPool = new CognitoUserPool(poolId, clientId, provider);
            var user = new CognitoUser(username, clientId, userPool, provider);

            var authParameters = new Dictionary<string, string>
            {
                { "USERNAME", username },
                { "PASSWORD", password }
            };

            var startAuthRequest = new InitiateSrpAuthRequest
            {
                AuthParameters = authParameters,
                ClientId = clientId
            };

            var authResponse = await user.StartWithSrpAuthAsync(startAuthRequest).ConfigureAwait(false);

            var tokens = authResponse.AuthenticationResult;
            var accessToken = tokens.AccessToken;

            Console.WriteLine("JWT Bearer Token: " + accessToken);
        }
    }
}
```

But the above code gives the error
```
error CS0117: 'InitiateSrpAuthRequest' does not contain a definition for 'AuthParameters'
error CS0117: 'InitiateSrpAuthRequest' does not contain a definition for 'ClientId'
```

I don't seem to understand what is the issue here and how to resolve this. Can anyone please let me know the problem here?

Thanks in advance.

**Comment by ashishdhingra** [comment, marked as answer]
@Sourabh-ALTEN The class definition for [InitiateSrpAuthRequest](https://github.com/aws/aws-sdk-net-extensions-cognito/blob/master/src/Amazon.Extensions.CognitoAuthentication/CognitoAuthenticationClasses.cs#L76) does not define the fields `AuthParameters` and `ClientId`. `AuthParameters` field appears to be defined in [InitiateCustomAuthRequest](https://github.com/aws/aws-sdk-net-extensions-cognito/blob/08b72052251eb148564b3d6ce3ddc616a4351036/src/Amazon.Extensions.CognitoAuthentication/CognitoAuthenticationClasses.cs#LL108C18-L108C43). I'm unsure about the source for your example, but kindly refer [Amazon CognitoAuthentication extension library examples](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/cognito-authentication-extension.html) for guidance on how to use this library.

Thanks,
Ashish

**Reply by Sourabh-ALTEN** [reply]
Hello @ashishdhingra , yes you are correct. I overlooked the class definition. Thank you for the update

[a comment by an automated account is not shown]
